docs/apim/4.0/guides/gravitee-expression-language.md

```diff
@@ -193,6 +193,17 @@ Request context attributes and examples are listed below.
 {% endtab %}
 {% endtabs %}
 
+#### Important Note for v4 Emulation Engine Users
+When using the v4 emulation engine, it is important to note that the api-key attribute is no longer accessible in the context if you wish to use it as a logging filter condition. Instead of referencing the api-key attribute directly with expressions like `{#context.attributes['api-key'] == 'api-key-value'}`, users can retrieve the API key from request parameters or headers.
+
+To achieve this, consider using the following expression:
+
+```
+{(#request.params['api-key'] != null && #request.params['api-key'][0] == 'my-api-key') || #request.headers['X-Gravitee-Api-Key'][0] == 'my-api-key'}
+```
+
+This approach checks for the API key either in the query parameters (api-key) or in the headers (X-Gravitee-Api-Key), ensuring compatibility with v4 emulation.
```

Comment:
I would change "ensuring compatibility with v4 emulation." to "which ensures compatibility with v4 emulation."